tests/test_interpreter.py

```diff
@@ -2,32 +2,33 @@
 from random import randint
 import time
 import pytest
-import interpreter
+import interpreter as i
 from interpreter.utils.count_tokens import count_tokens, count_messages_tokens
 import time
 
-interpreter.auto_run = True
-interpreter.model = "gpt-4"
-interpreter.temperature = 0
-
 
 # this function will run before each test
 # we're clearing out the messages Array so we can start fresh and reduce token usage
-def setup_function():
+
+@pytest.fixture(scope="function") # This will make the interpreter instance available to all test cases.
+def interpreter():
+    interpreter = i.create_interpreter()
```

Comment:
Does this mean that every test case is using a new instance of interpreter?

One of the things we were doing previously was testing that clearing messages via the `.reset()` method worked as expected.